src/api/namespaces/mutate/deleteNamespace.ts

```diff
@@ -0,0 +1,65 @@
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+
+import { NamespaceDeletedSchema } from "../schema";
+import { apiFactory } from "~/api/apiFactory";
+import { namespaceKeys } from "..";
+import { useApiKey } from "~/util/store/apiKey";
+import { useToast } from "~/design/Toast";
+import { useTranslation } from "react-i18next";
+
+const deleteNamespace = apiFactory({
+  url: ({ namespace }: { namespace: string }) =>
+    `/api/namespaces/${namespace}?recursive=true`,
+  method: "DELETE",
+  schema: NamespaceDeletedSchema,
+});
+
+type ResolvedCreateNamespace = Awaited<ReturnType<typeof deleteNamespace>>;
+
+export const useDeleteNamespace = ({
+  onSuccess,
+}: { onSuccess?: (data: ResolvedCreateNamespace) => void } = {}) => {
+  const apiKey = useApiKey();
+  const { toast } = useToast();
+  const queryClient = useQueryClient();
+  const { t } = useTranslation();
+
+  return useMutation({
+    mutationFn: ({ namespace }: { namespace: string }) =>
+      deleteNamespace({
+        apiKey: apiKey ?? undefined,
+        urlParams: {
+          namespace,
+        },
+      }),
+    onSuccess(data, variables) {
+      /**
+       * invalidating the cache is important here, because after deleting the namespace
+       * we will redirect to the frontpage, where we pick the first namespace we can
+       * find and redirect to it. It is very likely that the cache will be used here
+       * (namespace cache gets populated very ealy in the app lifecycle), so we need
+       * to make sure that we don't accidentally redirect to the namespace we just
+       * deleted.
+       */
+      queryClient.invalidateQueries(namespaceKeys.all(apiKey ?? undefined));
```

Comment:
I wasn't sure about the right approach here but I think I was able to clarify it for myself, and I think the current implementation is good. Just sharing my thoughts FYI:

* Would it be feasible and sufficient to just delete the specific namespace from cache that we have just deleted?
* On the other hand, I think "single source of truth" is a good principle to follow here, except...
* ...what if deleting a namespace takes longer (because it's a cascading process) and the API still lists the namespace at this point in time? Well, I tested this and it doesn't, so that seems to be fine.

Reply:
> Would it be feasible and sufficient to just delete the specific namespace from cache that we have just deleted?

Yes, this would work as well, I think I wanted to keep it simple because this one is harder to test and just flushing the cache altogether will make it a bit more predictable. "Single source of truth" is a nice framing for that!

> ...what if deleting a namespace takes longer (because it's a cascading process) and the API still lists the namespace at this point in time? Well, I tested this and it doesn't, so that seems to be fine.

- I'm pretty sure the namespace is (marked as) deleted when the api call resolves, and some background tassk will clean up after that.